File: Python/maximum_sum_subarray.py
def max_sum_subarray(array):
    total, answer = array[0], array[0]
    for i in range(1, len(array)):
        total = max(array[i], total + array[i])
        answer = max(answer, total)
    return answer


print(max_sum_subarray([1, -3, 2, -5, 7, 6, -1, -4, 11, -23]))

File: Java/MaximumSumSubarray.java
public class MaximumSumSubarray {
    public static void main(String[] args) {
        int[] nums = {1, -3, 2, -5, 7, 6, -1, -4, 11, -23};
        System.out.println(maximumSumSubarray(nums));
    }

    private static int maximumSumSubarray(int[] nums) {
        int answer = nums[0], sum = nums[0];
        for (int i = 1; i < nums.length; i++) {
            sum = Math.max(sum + nums[i], nums[i]);
            answer = Math.max(answer, sum);
        }
        return answer;
    }
}